app/src/main/java/com/ml/shubham0204/docqa/ui/screens/docs/DocsViewModel.kt

```diff
@@ -55,6 +59,38 @@ class DocsViewModel(
         }
     }
 
+    suspend fun addDocumentFromUrl(url: String, context: Context, callback: (Boolean) -> Unit) = withContext(Dispatchers.IO){
+        try {
+            val connection = URL(url).openConnection() as HttpURLConnection
+            connection.connect()
+            if (connection.responseCode == HttpURLConnection.HTTP_OK) {
+                val inputStream = connection.inputStream
+                val fileName = url.substringAfterLast("/")
```

Comment:
Also, the URL may contain the file name in diverse ways. This [SO answer](https://stackoverflow.com/a/11576046/13546426) provides a robust function that can manage multiple cases.